src/Kmd.Logic.Gateway.Automation/Publish.cs

```diff
@@ -133,18 +133,26 @@ private async Task CreateOrUpdateProducts(Guid subscriptionId, Guid providerId,
         {
             if (entityId.HasValue)
             {
+                var policies = await client.GetAllPoliciesAsync(subscriptionId, entityId.Value, entityType).ConfigureAwait(false);
+
                 if (rateLimitPolicy != null)
                 {
                     var rateLimitPolicyRequest = new RateLimitPolicyRequest(
-                        rateLimitPolicy.Name, entityId, entityType, rateLimitPolicy.Description, rateLimitPolicy.Calls, rateLimitPolicy.RenewalPeriod);
+                     rateLimitPolicy.Name, entityId, entityType, rateLimitPolicy.Description, rateLimitPolicy.Calls, rateLimitPolicy.RenewalPeriod);
                     switch (policiesResults.RateLimitPolicy.Status)
                     {
                         case ValidationStatus.CanBeCreated:
                             var created = await client.CreateRateLimitPolicyAsync(subscriptionId, rateLimitPolicyRequest).ConfigureAwait(false);
                             this.publishResults.Add(new GatewayAutomationResult() { ResultCode = ResultCode.RateLimitPolicyCreated, EntityId = created.Id });
                             break;
                         case ValidationStatus.CanBeUpdated:
-                            // TODO: Update RateLimitPolicy
+                            var ratelimitPolicyDoc = policies.FirstOrDefault(p => p.PolicyType == "RateLimit");
```

Comment:
You already have rate limit result. You already know whether you should update or not.

Reply:
Fixed, fetching from validation result now .